spring-integration-sftp/src/main/java/org/springframework/integration/sftp/session/DefaultSftpSessionFactory.java

```diff
@@ -583,18 +594,18 @@ public void showMessage(String message) {
 		@Override
 		public String[] promptKeyboardInteractive(String destination, String name, String instruction, String[] prompt,
 				boolean[] echo) {
-			if (hasDelegate()) {
-				if (getDelegate() instanceof UIKeyboardInteractive) {
-					return ((UIKeyboardInteractive) getDelegate()).promptKeyboardInteractive(destination, name,
-							instruction, prompt, echo);
-				}
+			if (hasDelegate() && getDelegate() instanceof UIKeyboardInteractive) {
+				return ((UIKeyboardInteractive) getDelegate()).promptKeyboardInteractive(destination, name,
+						instruction, prompt, echo);
 			}
-			if (logger.isDebugEnabled()) {
-				logger.debug("No UserInfo provided - " + destination + ":" + name + ":" + instruction + ":"
-						+ Arrays.asList(prompt) + ":" + Arrays.asList(echo));
+			else {
+				if (logger.isDebugEnabled()) {
+					logger.debug("No UIKeyboardInteractive provided - " + destination + ":" + name + ":" + instruction
+							+ ":" + Arrays.asList(prompt) + ":" + Arrays.asList(echo));
+				}
+				return null;
 			}
-			return null;
 		}
 	}
 
-}
+}
```

Comment:
Minor: new line in the end of file.
Looks like you have used a different device for this commit or a fresh version of your IDE.
Anyway settings may be broken.